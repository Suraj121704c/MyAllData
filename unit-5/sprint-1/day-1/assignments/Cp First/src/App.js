import "./App.css";
import Counter from "./components/Counter";

function App() {
  return (
    <div className="App">
      <div data-testid="counter">{/* counter component */}
      <Counter />
      </div>
    </div>
  );
}

export default App;
