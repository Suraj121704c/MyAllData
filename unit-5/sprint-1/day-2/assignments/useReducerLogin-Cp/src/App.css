.App {
  text-align: center;
}

.form-wrapper {
  width: fit-content;
  margin: auto;
}

.useremail-wrapper,
.userpassword-wrapper {
  width: 500px;
  display: flex;
  margin: 0.5rem 0;
}

.useremail-wrapper label,
.userpassword-wrapper label {
  width: 200px;
  font-family: "Poppins", sans-serif;
  font-size: 20px;
}
