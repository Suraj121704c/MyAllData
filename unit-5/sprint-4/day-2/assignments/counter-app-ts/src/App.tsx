import React, { useEffect, useState } from "react";
function App() {
  const [count, setCount] = useState<number>(0);

  useEffect(() => {
    const storedCount = localStorage.getItem("counter");
    if (storedCount !== null) {
      setCount(Number(storedCount));
    }
  }, []);

  useEffect(() => {
    localStorage.setItem("counter", count.toString());
  }, [count]);

  const handleIncrement = () => {
    setCount(count + 1);
  };

  const handleDecrement = () => {
    setCount(count - 1);
  };

  const handleIncrement5 = () => {
    setCount(count + 5);
  };

  const handleDecrement5 = () => {
    setCount(count - 5);
  };

  return (
    <div className="App">
      <h1 data-testid="counter">{count}</h1>
      <button data-testid="increment" onClick={handleIncrement}>
        Increment
      </button>
      <button data-testid="decrement" onClick={handleDecrement}>
        Decrement
      </button>
      <button data-testid="increment5" onClick={handleIncrement5}>
        Increment 5 times
      </button>
      <button data-testid="decrement5" onClick={handleDecrement5}>
        Decrement 5 times
      </button>
    </div>
  );
}

export default App;
