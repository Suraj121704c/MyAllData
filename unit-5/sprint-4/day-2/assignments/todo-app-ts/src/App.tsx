import React from "react";
import Code from "./Components/Code";
function App() {
  return (
    <div className="App">
      <Code />
    </div>
  );
}

export default App;
