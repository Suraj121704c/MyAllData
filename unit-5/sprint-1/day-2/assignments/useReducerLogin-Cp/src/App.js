import { useState } from "react";
import "./App.css";
import { useReducer } from "react";

const initialState = {
  email: "",
  password: "",
};

//should have the cases "email", "password", and "reset", along with the default cases
const reducer = (state, action) => {
  switch (action.type) {
    case "email": {
      return { ...state, email: action.payload };
    }
    case "password": {
      return { ...state, password: action.payload };
    }
    case "reset": {
      return initialState;
    }
    default: {
      return state;
    }
  }
};

function App() {
  // import and use the useReducer hook here, with the reducer function and the initialState.
  const [state, dispatch] = useReducer(reducer, initialState);

  //store the data in this object variable when you click on the submit button, to render, the data in the UI.
  const [submittedData, setSubmittedData] = useState({});

  const handleSubmit = (e) => {
    e.preventDefault();
    setSubmittedData(state);
    dispatch({
      type: "reset",
      payload: {
        ...initialState,
      },
    });
  };

  //  console.log(submittedData?.state?.email)

  // console.log(submittedData?.state?.email?.length);

  return (
    <div className="App">
      <h2>useReducer Hook</h2>
      <form
        className="form-wrapper"
        data-testid="form-wrapper"
        onSubmit={handleSubmit}
      >
        <div className="useremail-wrapper">
          <label>User Email</label>
          <input
            type="email"
            data-testid="user-email"
            value={state.email}
            onChange={(e) =>
              dispatch({ type: "email", payload: e.target.value })
            }
          />
        </div>
        <div className="userpassword-wrapper">
          <label>User Password</label>
          <input
            type="password"
            data-testid="user-password"
            value={state.password}
            onChange={(e) =>
              dispatch({ type: "password", payload: e.target.value })
            }
          />
        </div>
        <button type="submit">Submit</button>
      </form>

      {submittedData.email && submittedData.password ? (
        <div>
          <div data-testid="submitted-data-email">
            User Email:{submittedData.email}
          </div>
          <div data-testid="submitted-data-password">
            User Password:{submittedData.password}
          </div>
        </div>
      ) : (
        <div data-testid="no-details-container">no details found</div>
      )}
    </div>
  );
}

// DO NOT change/modify the exports
export default App;
export { reducer, initialState };
