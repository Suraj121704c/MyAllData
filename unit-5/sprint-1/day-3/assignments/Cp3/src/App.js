import "./App.css";
import useTimeout from "./hooks/useTimeout";
import useOnlineStatus from "./hooks/useOnlineStatus";

function App() {
  // bring useTimeout hook here
  const { ready } = useTimeout(5000);
  // bring useOnlineStatus hook here
  const { isUserOnline } = useOnlineStatus();

  return (
    <div className="App">
      <h1>Custom Hooks</h1>

      <h3>Timeout</h3>
      {ready && <div data-testid="timeout-div">After the timeout</div>}

      <br />
      <h3>Online Status</h3>

      {/* If the user's network status is online, show this div only */}

      {/* If the user's network status is offline, show this div only */}

      <div>
        {isUserOnline ? (
          <div data-testid="online-status">Online ðŸŸ¢</div>
        ) : (
          <div data-testid="offline-status">Offline ðŸ”´</div>
        )}
      </div>
    </div>
  );
}

export default App;
