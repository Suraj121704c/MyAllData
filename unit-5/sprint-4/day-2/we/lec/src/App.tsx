import React from "react";
import "./App.css";
import { TodoApp } from "./Components/TodoApp";

function App() {
  return (
    <div className="App">
      <h1>Type Script TodoApp</h1>
      <TodoApp />
    </div>
  );
}

export default App;
