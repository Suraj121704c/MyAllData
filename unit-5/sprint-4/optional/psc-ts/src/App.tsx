import React from "react";
import "./App.css";
import Clock from "./components/Clock";
import useClock from "./hooks/useClock";
import List1 from "./components/List1";
import List2 from "./components/List2";

function App() {
  const { hours, minutes, seconds } = useClock();
  return (
    <div className="App">
      {/* Clock */}
      <Clock hours={hours} minutes={minutes} seconds={seconds} />
      {/* List 1  initialValues [1, 2, 3] */}
      {/* List 2  initialValues [4, 5] */}
      <List1 list={[1,2,3]}/>
      <List2 list={[4,5]}/>
    </div>
  );
}

export default App;
