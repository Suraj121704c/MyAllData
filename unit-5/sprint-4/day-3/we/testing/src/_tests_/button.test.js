import { fireEvent, render, screen } from "@testing-library/react";
import Button from "../Components/Button";
import App from "../App";
// import Counter from "../Components/Counter";
import renderer from "react-test-renderer";

// if u wanna check first off all render it
// screen.debug use to show the code
describe("Tesing Button Component", () => {
  it("Button should render on DOM", () => {
    render(
      <Button color={"teal"} size={"large"}>
        Click Me
      </Button>
    );
    // screen.debug();
    const button = screen.getByText("Click Me");
    expect(button).toBeInTheDocument();
  });
  // checking by id
  // getAllByTestId takes array
  // getByTestId takes one id
  it("Button should be present on DOM", () => {
    render(<Button>Custom Button</Button>);
    const button = screen.getByTestId("custom-button");
    expect(button).toBeInTheDocument();
  });
  // checking multiple buttons
  it("Check for Button text Content", () => {
    render(<App />);
    const buttons = screen.getAllByTestId("custom-button");
    expect(buttons[0]).toHaveTextContent("Click Me");
    expect(buttons[1]).toHaveTextContent("Click");
  });

  it("Should invoke the function", () => {
    const mockFunc = jest.fn();
    render(<Button onClick={mockFunc} />);
    const button = screen.getByTestId("custom-button");
    fireEvent.click(button);
    fireEvent.click(button);
    expect(mockFunc).toBeCalledTimes(2);
  });
  // used for snapshot testing
  it("Should snapshot", () => {
    const tree = renderer
      .create(
        <Button color={"blue"} size={"large"}>
          Click Me
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
