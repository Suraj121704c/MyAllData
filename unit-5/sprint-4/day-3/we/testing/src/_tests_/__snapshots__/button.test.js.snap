// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tesing Button Component Should snapshot 1`] = `
<div>
  <button
    className="blue large"
    data-testId="custom-button"
  >
    Click Me
  </button>
</div>
`;
